executor/src/witgen/jit/compiler.rs

```diff
@@ -0,0 +1,528 @@
+use std::{iter, mem, sync::Arc};
+
+use auto_enums::auto_enum;
+use itertools::Itertools;
+use libloading::Library;
+use powdr_number::FieldElement;
+
+use crate::witgen::{jit::affine_symbolic_expression::MachineCallArgument, machines::LookupCell};
+
+use super::{
+    affine_symbolic_expression::{Assertion, Effect},
+    cell::Cell,
+    symbolic_expression::{BinaryOperator, SymbolicExpression, UnaryOperator},
+};
+
+#[repr(C)]
+pub struct WitgenFunctionParams<T> {
+    data: *mut T,
+    known: *mut u32,
+    len: u64,
+    row_offset: u64,
+    call_machine: extern "C" fn(u64, *mut LookupCell<'_, T>, usize) -> bool,
+}
+
+pub type WitgenFunction<T> = extern "C" fn(WitgenFunctionParams<T>);
+
+pub fn compile_effects<T: FieldElement>(
+    first_column_id: u64,
+    column_count: usize,
+    known_inputs: &[Cell],
+    effects: &[Effect<T, Cell>],
+) -> Result<(Arc<Library>, WitgenFunction<T>), String> {
```

Comment:
We need to keep the `Library` in an Arc because otherwise it will be unloaded. We can change the interface once we know where and how the loaded functions will be stored.

Reply:
Do I understand this right, the returned library needs to stay in the scope as long as the function might be used? If yes, please add this to the docstring :)

Reply:
Fixed